wlm/src/main/java/ai/djl/serving/wlm/LmiUtils.java

```diff
@@ -45,67 +44,42 @@ public final class LmiUtils {
 
     private static final Logger logger = LoggerFactory.getLogger(LmiUtils.class);
 
-    private static final List<String> DEEPSPEED_MODELS =
-            List.of(
-                    "bert",
-                    "bloom",
-                    "gpt_neo",
-                    "gpt_neox",
-                    "gpt2",
-                    "gptj",
-                    "opt",
-                    "roberta",
-                    "xlm-roberta");
-
     private LmiUtils() {}
 
     static String inferLmiEngine(ModelInfo<?, ?> modelInfo) throws ModelException {
-        // MMS/Torchserve
-        if (Files.isDirectory(modelInfo.modelDir.resolve("MAR-INF"))) {
-            logger.info("Found legacy torchserve model, use Python engine.");
-            return "Python";
-        }
-
-        Properties prop = modelInfo.prop;
+        Properties prop = modelInfo.getProperties();
         HuggingFaceModelConfig modelConfig = getHuggingFaceModelConfig(modelInfo);
         if (modelConfig == null) {
-            String engineName = isTrtLLM(modelInfo) ? "MPI" : "Python";
+            String engineName = isTrtLLM(prop) ? "MPI" : "Python";
             logger.info("No config.json found, use {} engine.", engineName);
             return engineName;
         }
-        String features = Utils.getenv("SERVING_FEATURES");
-        String modelType = modelConfig.getModelType();
-        String engineName;
-        if ("stable-diffusion".equals(modelType)) {
-            // TODO: Move this from hardcoded to deduced in PyModel
-            prop.setProperty("option.entryPoint", "djl_python.stable-diffusion");
-            engineName = "DeepSpeed";
-        } else if (isDeepSpeedRecommended(modelType)) {
-            engineName = "DeepSpeed";
-        } else if (features != null && features.contains("trtllm")) {
-            engineName = "MPI";
-        } else {
-            engineName = "Python";
-        }
-        logger.info("Detected engine: {}, modelType: {}", engineName, modelType);
-        return engineName;
+        LmiConfigRecommender.configure(prop, modelConfig);
```

Comment:
did we ensure we only enter here when customer provide something like option.model_id and we know it is LLM. This is just to prepare for use case on us to support something like

```
HF_MODEL_ID=bge-word-embedding
```

word embedding model doesn't need TP=max, it should be TP1 instead. We can change it in the LMI_config chapter. If model is identified as LLM, then we do TP=max

Reply:
We only enter this recommender if we determine that we are using a "Python" model, and we can fetch the HuggingFace Model Config. While not perfect, that seems like a pretty reasonable requirement to enter this `auto-configure` behavior.

For embedding models, those will also have a HuggingFace Model Config and maybe that contains some information we can use (like architecture or task) to execute separate behavior here. We might also want to do some more analysis here (such as checking model architecture is CausalLM or Seq2SeqLM). We can probably follow up here once we have a better grasp on how to separate generation use-case and embedding use-case